experimental/Schemes/singular_locus.jl

```diff
@@ -3,28 +3,121 @@ export singular_locus, singular_locus_reduced
 export reduced_scheme
 export is_smooth
 
-MPAnyQuoRing = Union{MPolyQuoLocalizedRing, 
-                MPolyQuo
-               }
 
-MPAnyNonQuoRing = Union{MPolyRing, MPolyLocalizedRing
-                  }
+#######################################################################
+# Functionality for smoothness/singular locus
+#######################################################################
 
-### TODO: The following two functions need to be made type-sensitive 
+#######################################################################
+# MOVE TO Methods
+#######################################################################
+
+### TODO: The following two functions (singular_locus, 
+###       singular_locus_reduced) need to be made type-sensitive 
 ###       and reduced=true needs to be set automatically for varieties 
 ###       as soon as not only schemes, but also varieties as special 
 ###       schemes have been introduced in OSCAR
-function singular_locus(X::AbsSpec{<:Ring, <:MPAnyQuoRing})
+### TODO: Make singular locus also available for projective schemes and
+###       for covered schemes (the workhorse is already there...).
+ 
+@doc Markdown.doc"""
+    singular_locus(X::AbsSpec{<:Field, <:MPolyAnyRing}) -> AbsSpec
+
+Return the singular locus of `X`.
+
+For computing the singular locus of the reduced scheme induced by `X`,
+please use [`singular_locus_reduced`](@ref).
+
+Currently this command is available for affine schemes and space germs.
+TODO: Covered schemes, projective schemes
```

Comment:
Eventually the documentation should then be moved to a dummy routine `singular_locus(X::Scheme)` which does nothing but reporting an error but can be used to carry a docstring.

Reply:
This is precisely where this should be heading in the end.

Reply:
By the way, do we already have this Scheme as a type or union yet?

Reply:
Yes, we do.